docs/type.rst

```diff
@@ -76,24 +110,64 @@ Default integer and float-point precisions (``default_ip`` and ``default_fp``) c
 Type casts
 ----------
 
-Use ``ti.cast`` to cast scalar values.
+Implicit casts
+****************
+
+WARNING: The type of a variable is **determinated on it's initialization**.
+
+When a *wide* variable is assigned with a *narrow* type, it will be
+implicitly promoted to the *wide* type and no warning will be raised:
 
 .. code-block:: python
 
-    a = 1.4
-    b = ti.cast(a, ti.i32)
-    c = ti.cast(b, ti.f32)
+    a = 1.7
+    a = 1
+    print(a)  # 1.0
 
-    # Equivalently, use ``int()`` and ``float()``
-    # to convert values to default float-point/integer types
-    b = int(a)
-    c = float(b)
+When a *narrow* variable is assigned with a *wide* type, it will be
+implicitly casted into the *narrow* type and Taichi will raise a warning:
 
-    # Element-wise casts in matrices
-    mat = ti.Matrix([[3.0, 0.0], [0.3, 0.1]])
-    mat_int = mat.cast(int)
-    mat_int2 = mat.cast(ti.i32)
+.. code-block:: python
+
+    a = 1
+    a = 1.7
+    print(a)  # 1
+
+Explicit casts
+**************
+
+You may use ``ti.cast`` to explicitly cast scalar values between different types:
+
+.. code-block:: python
+
+    a = 1.7
+    b = ti.cast(a, ti.i32)  # 1
+    c = ti.cast(b, ti.f32)  # 1.0
+
+Equivalently, use ``int()`` and ``float()`` to convert values to default float-point/integer types:
+
+.. code-block:: python
+
+    a = 1.7
+    b = int(a)    # 1
+    c = float(a)  # 1.0
+
+Casting vector / matrix elements
+********************************
+
+Type cast applied to vectors / matrices are element-wise:
```

Comment:
I think the slash here doesn't need the spaces before and after. But please don't quote me here until we get opinions from a native speaker.
```suggestion
Type casts applied to vectors/matrices are element-wise:
```